components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/DBUtil.java

```diff
@@ -0,0 +1,120 @@
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInput;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class DBUtil {
+
+    private static Log log = LogFactory.getLog(SAMLSSOUtil.class);
+
+    private static final String SAML2_ASSERTION_STORE = "IDN_SAML2_ASSERTION_STORE";
+    private static final String ASSERTION = "ASSERTION";
+
+    private static boolean isAssertionDTOPersistenceSupported = false;
+    private static boolean isAssertionDTOPersistenceStatusChecked = false;
+
+    private DBUtil() {}
+
+    /**
+     * Return if IDN_SAML2_ASSERTION_STORE table has ASSERTION column.
+     *
+     * @return Existence of ASSERTION column in IDN_SAML2_ASSERTION_STORE
+     */
+    public static boolean isAssertionDTOPersistenceSupported() {
+
+        if(!isAssertionDTOPersistenceStatusChecked) {
+            try (Connection connection = IdentityDatabaseUtil.getDBConnection()) {
+
+                DatabaseMetaData metaData = connection.getMetaData();
+                ResultSet rs = metaData.getColumns(null, null, SAML2_ASSERTION_STORE, ASSERTION);
+                if (rs.next()) {
+                    isAssertionDTOPersistenceSupported = true;
+                }
+            } catch (SQLException e) {
+                log.error("Error in fetching metadata from IDN_SAML2_ASSERTION_STORE database", e);
+            }
+            isAssertionDTOPersistenceStatusChecked = true;
+        }
+        return isAssertionDTOPersistenceSupported;
+    }
+
+    /**
+     * Retun java object from input stream. Used to retrieve blob objects from database.
+     *
+     * @param is Input stream.
+     * @return Java object constructed from the input stream.
+     * @throws IOException
+     * @throws ClassNotFoundException
+     */
+    public static Object getBlobObject(InputStream is) throws IOException, ClassNotFoundException {
+
+        if (is != null) {
+            ObjectInput ois = null;
+            try {
+                ois = new ObjectInputStream(is);
+                return ois.readObject();
+            } finally {
+                if (ois != null) {
+                    ois.close();
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Serialize an object and set into a prepared statement as a blob.
+     *
+     * @param prepStmt Prepared statement.
+     * @param value    Object to be saved.
+     * @param index    Index of the prepared statement.
+     * @throws SQLException
+     * @throws IOException
+     */
+    public static void setBlobObject(PreparedStatement prepStmt, Object value, int index) throws SQLException, IOException {
```

Comment:
Can you check whether clob/text could be used instead of blob?

Reply:
Blob and Clob have almost the same performance and the only difference is clob is collation and encoding sensitive. and clob has not been used anywhere in the current DB scripts. shall we proceed with blobs